ownCloud/UI Elements/LockWindow.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  LockWindow.swift
+//  ownCloud
+//
+//  Created by Javier Gonzalez on 17/05/2018.
+//  Copyright © 2018 ownCloud GmbH. All rights reserved.
+//
+
+/*
+ * Copyright (C) 2018, ownCloud GmbH.
+ *
+ * This code is covered by the GNU Public License Version 3.
+ *
+ * For distribution utilizing Apple mechanisms please see https://owncloud.org/contribute/iOS-license-exception/
+ * You should have received a copy of this license along with this program. If not, see <http://www.gnu.org/licenses/gpl-3.0.en.html>.
+ *
+ */
+
+import UIKit
+
+//Subclass that only allows the protrait mode
+class LockWindow: UIWindow {
```

Comment:
Actually after seeing more or less all the code I don't see this class necessary, the only thing that you do here that differentiate this class from its parent is the presentation, you can do this in an extension or in the class you call it.  This way you can get rid of this class.

Reply:
This is something that in the future could happen to us again so, maybe, worth create an `extension` to `NSObject` or to `Object` assigning it an identifier. This way we can reuse the identifier in more situations we'll need in the future and we can get rid of this class.

Reply:
I see it is a different way to do the same. I follow the indications that @felix-schwarz told me and I think that the current approach it is ok.